### PR TITLE
[RELAY]reshape_like

### PR DESCRIPTION
**Pull request by siju-samuel**
#1799

Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from others in the community.

### CODE REVIEW

File: src/relay/ir/type.cc [hunk 1]
```diff
@@ -22,6 +22,14 @@ TensorType TensorTypeNode::Scalar(DataType dtype) {
   return TensorTypeNode::make({}, dtype);
 }
 
+IndexExpr TensorTypeNode::Size() const {
+  IndexExpr size = make_const(Int(64), 1);
```
Comment (tqchen) [inline src/relay/ir/type.cc]: directly product from shape[0], when the shape.size() equals 0, return make_const(Int(32), 1);

**Comment by tqchen** [comment]
@siju-samuel please request reviews from https://github.com/dmlc/tvm/blob/master/CONTRIBUTORS.md#reviewers

**Comment by siju-samuel** [comment]
@MarisaKirisame @slyubomirsky @nishi-t @junrushao1994 could you please help to review this PR. Thanks.

File: include/tvm/relay/type.h [hunk 2]
```diff
@@ -82,6 +82,9 @@ class TensorTypeNode : public BaseTensorTypeNode {
     v->Visit("span", &span);
   }
 
+  /*! \brief Return product of elements in the shape */
```
Comment (jroesch) [inline include/tvm/relay/type.h]: Could we add one more line to the docs such as `(d_1, d_2, ..., d_n) -> d1 * d_2 ... * d_n`.

**Comment by tqchen** [comment]
Thanks @siju-samuel @MarisaKirisame @jroesch this is now merged

**Pull request merged**